pkg/block/fetcher.go

```diff
@@ -428,16 +428,33 @@ func NewDeduplicateFilter() *DeduplicateFilter {
 // Filter filters out duplicate blocks that can be formed
 // from two or more overlapping blocks that fully submatches the source blocks of the older blocks.
 func (f *DeduplicateFilter) Filter(metas map[ulid.ULID]*metadata.Meta, synced GaugeLabeled, _ bool) {
-	root := NewNode(&metadata.Meta{
-		BlockMeta: tsdb.BlockMeta{
-			ULID: ulid.MustNew(uint64(0), nil),
-		},
-	})
+	var wg sync.WaitGroup
 
-	metaSlice := []*metadata.Meta{}
+	metasByResolutionMap := make(map[int64][]*metadata.Meta)
```

Comment:
Go is a language that is type safe, so we don't need add suffix to name what `type` this variable is. IDE and code already show us that this is a `map`. No need for this suffix.